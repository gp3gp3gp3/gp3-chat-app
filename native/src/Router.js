import React, { Component } from 'react'
import { connect } from 'react-redux'
import firebase from 'firebase'
import { View, StyleSheet } from 'react-native'
import Login from './components/Login'

class Router extends Component {
  constructor (props) {
    super(props)
    const config = {
      apiKey: 'AIzaSyBHymkx-sx64eGJFLNmv8Y7nwakdXCNcpQ',
      authDomain: 'gp3-chat-app.firebaseapp.com',
      databaseURL: 'https://gp3-chat-app.firebaseio.com',
      projectId: 'gp3-chat-app',
      storageBucket: '',
      messagingSenderId: '111539696639'
    }
    firebase.initializeApp(config)
    // firebase
    //   .auth()
    //   .onAuthStateChanged(user => {
    //     if (user) {
    //       props.login(user)
    //     }
    //   })
  }

  render () {
    return (
      <Login />
    )
  }
}

const mapStateToProps = ({ auth: { authenticated } }) => {
  return { authenticated }
}

export default connect(mapStateToProps)(Router)
