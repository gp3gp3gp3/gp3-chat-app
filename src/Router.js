import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import { connect } from 'react-redux'
import firebase from 'firebase'
import Header from './components/Header/Header'
import MessageList from './containers/MessageList'
import MessageBox from './containers/MessageBox'
import Login from './containers/Login'
import { login } from './actions'
import './App.css'

class Router extends Component {
  constructor (props) {
    super(props)
    const config = {
      apiKey: 'AIzaSyBHymkx-sx64eGJFLNmv8Y7nwakdXCNcpQ',
      authDomain: 'gp3-chat-app.firebaseapp.com',
      databaseURL: 'https://gp3-chat-app.firebaseio.com',
      projectId: 'gp3-chat-app',
      storageBucket: '',
      messagingSenderId: '111539696639'
    }
    firebase.initializeApp(config)
    firebase
      .auth()
      .onAuthStateChanged(user => {
        if (user) {
          props.login(user)
        }
      })
    this.scrollToBottom = this.scrollToBottom.bind(this)
  }

  scrollToBottom () {
    const node = ReactDOM.findDOMNode(this.refs['messageBox'])
    window.scrollTo({
      left: 0,
      top: node.offsetTop,
      behavior: 'smooth'
    })
  }

  renderApp () {
    if (this.props.loading) {
      return <div />
    } else if (this.props.authenticated) {
      return (
        <div className='Body'>
          <MessageList
            scrollToBottom={this.scrollToBottom}
          />
          <MessageBox
            ref='messageBox'
          />
        </div>
      )
    } else {
      return <Login />
    }
  }

  render () {
    return (
      <div className='App'>
        <Header />
        {this.renderApp()}
      </div>
    )
  }
}

const mapStateToProps = ({ auth: { authenticated, loading } }) => {
  return { authenticated, loading }
}

export default connect(mapStateToProps, { login })(Router)
