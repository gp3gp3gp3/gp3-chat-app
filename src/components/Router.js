import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import { connect } from 'react-redux'
import firebase from 'firebase'
import Header from '../containers/Header'
import MessageList from '../containers/MessageList'
import MessageBox from '../containers/MessageBox'
import Login from '../containers/Login'
import { login } from '../actions'
import './App.css'

export class Router extends Component {
  constructor (props) {
    super(props)
    const config = {
      apiKey: 'AIzaSyBHymkx-sx64eGJFLNmv8Y7nwakdXCNcpQ',
      authDomain: 'gp3-chat-app.firebaseapp.com',
      databaseURL: 'https://gp3-chat-app.firebaseio.com',
      projectId: 'gp3-chat-app',
      storageBucket: '',
      messagingSenderId: '111539696639'
    }
    firebase.initializeApp(config)
    firebase
      .auth()
      .onAuthStateChanged(user => {
        if (user) {
          props.login(user)
        }
      })
    this.scrollToBottom = this.scrollToBottom.bind(this)
  }

  scrollToBottom () {
    const node = ReactDOM.findDOMNode(this.refs['messageBox'])
    window.scrollTo({
      left: 0,
      top: node.offsetTop,
      behavior: 'smooth'
    })
  }

  renderApp () {
    if (this.props.authenticated) {
      return (
        <div className='Body'>
          <MessageList
            scrollToBottom={this.scrollToBottom}
          />
          <MessageBox
            ref='messageBox'
          />
        </div>
      )
    } else {
      return <Login />
    }
  }

  render () {
    return (
      <div className='App'>
        <Header />
        {this.renderApp()}
      </div>
    )
  }
}

const mapStateToProps = ({ auth: { authenticated } }) => {
  return { authenticated }
}

export default connect(mapStateToProps, { login })(Router)
