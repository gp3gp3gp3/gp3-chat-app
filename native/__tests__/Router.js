/* eslint-env jest */
import 'react-native'
// Note: test renderer must be required after react-native.
import { shallow } from 'enzyme'
import React from 'react'
import configureStore from 'redux-mock-store'
import Router from '../src/Router'

const middlewares = []
const mockStore = configureStore(middlewares)
const initialState = { auth: { authenticated: false } }

it('renders correctly with store', () => {
  const wrapper = shallow(
    <Router />,
    { context: { store: mockStore(initialState) } }
  )
  expect(wrapper).toMatchSnapshot()
})
