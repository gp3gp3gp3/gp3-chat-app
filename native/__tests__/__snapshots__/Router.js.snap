// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with store 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Router) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "authenticated": false,
      "dispatch": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "authenticated": false,
        "dispatch": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
